src/TodosContext.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useMemo, useState } from 'react';
+import { useLocalStorage } from './hooks/useLocalStorage';
+import { FilterMode } from './types/FilterMode';
+import { Todos } from './types/Todos';
+
+const intitialItems: Todos[] = [];
+const defoultKey = 'todos';
+
+type State = {
+  items: Todos[];
+  filterMode: FilterMode;
+  addItem: (value: string) => void;
+  deleteItem: (id: number) => void;
+  setFilterMode: (value: FilterMode) => void;
+  setItems: (newItems: Todos[]) => void;
+  getVisibleItems(mode: FilterMode, itemsList: Todos[],): Todos[];
+};
+
+const state: State = {
+  items: intitialItems,
+  filterMode: 'all',
+  addItem: () => { },
+  deleteItem: () => { },
+  setFilterMode: () => { },
+  setItems: () => { },
+  getVisibleItems: () => {
+    return intitialItems;
+  },
+};
+
+export const TodosContext = React.createContext(state);
```

Comment:
let's create here our own hook
```
export const useTodos = ():State => React.useContext(TodosContext);
```
it will help us to avoid redundant import

Reply:
Not fixed

Reply:
can u elaborate here?